submissions/dafen173/a-tiny-js-world2/index.js

```diff
@@ -0,0 +1,92 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+
+class Person {
+  constructor(gender, name, species) {
+    this.species = species;
+    this.gender = gender;
+    this.name = name;  
+  } 
+  setFriends(friends){
+    this.friends = friends;
+  } 
+  
+  sayAsAnimal(){
+    if (this.species === 'cat') {
+      this.speach = 'Meow';
+    } else if (this.species === 'cat-woman') {
+      this.speach = 'Meow'
+    } else {
+      this.speach = 'Wohw whow whow'
+    }
+  }
+}
+
+class Human extends Person {
+  constructor(gender, name, species){
+    super(gender, name, species);
+    this.species = species || 'human';
+    this.hands = 2;
+    this.legs = 2;
+    this.speach = 'Hello, people!';
+  }
+}
+
+class Animal extends Person {
+  constructor(gender, name, species){
+    super(gender, name, species);
+    this.hands = 0;
+    this.legs = 4;
```

Comment:
Whatever property is shared must be generalized in a common base class. This is one of OOP principles